lib/services/local/session.go

```diff
@@ -95,3 +97,249 @@ func (s *IdentityService) DeleteAllAppSessions(ctx context.Context) error {
 	}
 	return nil
 }
+
+// GetWebSession returns a web session state for the given user and session id
+func (s *IdentityService) GetWebSession(user, sessionID string) (services.WebSession, error) {
```

Comment:
Nit: Since you're refactoring all these methods anyway, might be a good excuse to enhance them to accept contexts? Feel free to disregard if that's too much work / out of scope.

Reply:
I'd consider this out of scope for now due to the amount of additional changes. Will revisit in the future.